### DISCUSSION TITLE
Announcing freetz-ble, a BLE Advertisement processor for freetz

### DISCUSSION THREAD

**Discussion by Ircama** [Freetz-NG/freetz-ng#952] [Allgemein]
[freetz-ble](https://github.com/Ircama/freetz-ble) is a simple tool to continuously collect [BLE](https://wikipedia.org/wiki/Bluetooth_Low_Energy) advertisements sent by Smart Home wireless sensors (like [Xiaomi Mijia BLE devices](https://esphome.io/components/sensor/xiaomi_ble.html)) through freetz and store data in clear form to a local DB.

The code is currently a proof of concept and needs special hardware (Telink TLSR8253 BLE 5.0 SoC module, serial USB interface, USB cable); an example is the [Ai-Thinker TB-03F-KIT module](https://docs.ai-thinker.com/_media/tb-03f-kit_specification_en.pdf).

More generally, in this mode freetz-ble can support BLE.

People interested in testing or contributing could check the [freetz-ble repository](https://github.com/Ircama/freetz-ble).

This post also completes the notes in the Q&A at https://github.com/Freetz-NG/freetz-ng/discussions/947#discussion-6190608.

**Comment by fda77** [comment]
You could add it as a normal package or library.
First you should add make/host-tools/XXX for the files you unpack to tools/dir. This XXX could then later be selected by the lib/pkg
